Library/Homebrew/formula.rb

```diff
@@ -413,24 +417,48 @@ def linked_keg
     Pathname.new("#{HOMEBREW_LIBRARY}/LinkedKegs/#{name}")
   end
 
-  def latest_head_prefix
+  def latest_head_version
+    return @latest_head_version if @latest_head_version
+
     head_versions = installed_prefixes.map do |pn|
       pn_pkgversion = PkgVersion.parse(pn.basename.to_s)
       pn_pkgversion if pn_pkgversion.head?
     end.compact
 
-    latest_head_version = head_versions.max_by do |pn_pkgversion|
+    @latest_head_version ||= head_versions.max_by do |pn_pkgversion|
       [Tab.for_keg(prefix(pn_pkgversion)).source_modified_time, pn_pkgversion.revision]
     end
+  end
+
+  def latest_head_prefix
     prefix(latest_head_version) if latest_head_version
   end
 
+  def latest_head_outdated?(options={})
+    return false unless latest_head_prefix
+
+    tab = Tab.for_keg(latest_head_prefix)
+    return true if stable && tab.stable_version && tab.stable_version < stable.version
+    return true if devel && tab.devel_version && tab.devel_version < devel.version
+
+    return false unless head && head.downloader.is_a?(VCSDownloadStrategy)
+
+    if options[:fetch_head]
+      downloader = head.downloader
+      downloader.shutup! unless ARGV.verbose?
+      return false unless downloader.fetch_last_commit
+      downloader.fetch unless downloader.local_commit?
+
+      !(head_pkg_version.version.commit.start_with?(latest_head_version.version.commit))
```

Comment:
Rethink a bit. The `start_with?` is kinda bother me. At the same time, I do want to keep prefix short. So can we try to use short commit in the prefix but store the full length prefix to tab file?